statics/js/topologymanager_package/src/topologymanager/topology_layout/skydive_default/index.ts

```diff
@@ -0,0 +1,129 @@
+import LayoutConfig from '../config';
+import * as events from 'events';
+import { DataManager } from '../base/index';
+import { LayoutBridgeUI, LayoutBridgeUII } from '../base/ui/index';
+import { LayoutUI, EdgeUI, NodeUI, GroupUI } from '../base/ui/index';
+import { LabelRetrieveInformationStrategy } from '../base/edge/label/index';
+
+export default class SkydiveDefaultLayout implements TopologyLayoutI {
+    uiBridge: LayoutBridgeUII;
+    dataManager: DataManager = new DataManager();
+    e: events.EventEmitter = new events.EventEmitter();
+    alias: string = "skydive_default";
+    active: boolean = false;
+    config: LayoutConfig;
+    dataSources: DataSourceRegistry = new DataSourceRegistry();
+    selector: string;
+    constructor(selector: string) {
+        this.selector = selector;
+        this.uiBridge = new LayoutBridgeUI(selector);
+        this.uiBridge.useEventEmitter(this.e);
+        this.uiBridge.useConfig(this.config);
+        this.uiBridge.useLayoutUI(new LayoutUI(selector));
+        this.uiBridge.useDataManager(this.dataManager);
+        this.uiBridge.useNodeUI(new NodeUI());
+        this.uiBridge.useGroupUI(new GroupUI());
+        this.uiBridge.useEdgeUI(new EdgeUI());
+        // this.uiBridge.setCollapseLevel(1);
+        // this.uiBridge.setMinimumCollapseLevel(1);
+        this.dataManager.useLayoutContext(this.uiBridge.layoutContext);
+    }
+    initializer() {
+        console.log("Try to initialize topology " + this.alias);
+        $(this.selector).empty();
+        this.active = true;
+    }
+    setCollapseLevel(collapseLevel: number) {
+        this.uiBridge.setCollapseLevel(collapseLevel);
+    }
+    setMinimumCollapseLevel(collapseLevel: number) {
+        this.uiBridge.setMinimumCollapseLevel(collapseLevel);
+    }
+    useLinkLabelStrategy(linkLabelType: string) {
+        const strategy = LabelRetrieveInformationStrategy(linkLabelType);
+        strategy.setup(this.config);
+        this.uiBridge.useLinkLabelStrategy(strategy);
+    }
+    useConfig(config: LayoutConfig) {
+        this.config = config;
+        this.uiBridge.useConfig(this.config);
+    }
+    remove() {
+        this.dataSources.sources.forEach((source: DataSourceI) => {
+            source.unsubscribe();
+        });
+        this.removeLayout();
+    }
+    removeLayout() {
+        this.active = false;
+        this.uiBridge.remove();
+        $(this.selector).empty();
+        this.dataManager.removeOldData();
+        this.e.emit('ui.update');
+    }
+    addDataSource(dataSource: DataSourceI, defaultSource?: boolean) {
+        this.dataSources.addSource(dataSource, !!defaultSource);
+    }
+    reactToDataSourceEvent(dataSource: DataSourceI, eventName: string, ...args: Array<any>) {
+        console.log('Skydive default layout got an event', eventName, args);
+        let e;
+        switch (eventName) {
+            case "SyncReply":
+                if (this.config.getValue('useHardcodedData')) {
```

Comment:
We should have hard coded topology. That is only for testing. There is already a way to inject topology to Skydive.